crates/iota-data-ingestion-core/src/executor.rs

```diff
@@ -24,6 +24,74 @@ use crate::{
 
 pub const MAX_CHECKPOINTS_IN_PROGRESS: usize = 10000;
 
+/// The Executor of the main ingestion pipeline process.
+///
+/// This struct orchestrates the execution of multiple worker pools, handling
+/// checkpoint distribution, progress tracking, and shutdown. It utilizes
+/// [`ProgressStore`] for persisting checkpoint progress and provides metrics
+/// for monitoring the indexing process.
+///
+/// # Docs
+/// For more info please check the [custom indexer docs](https://docs.iota.org/developer/advanced/custom-indexer).
```

Comment:
This `# Docs` section is somewhat unexpected. This whole block of comment represents the documentation for the item we are documenting and I would expect it to be self-contained.

If you think it is crucial to add such a reference I would suggest changing to 
```suggestion
/// For more info on the ingestion pattern supported by this library please check the [custom indexer docs](https://docs.iota.org/developer/advanced/custom-indexer).
```

But as such it might be more fit for the README (which is probably more detailed than the `docs.iota.org` page), or the `lib`. So you can also skip it altogether.

Reply:
I've opted to remove the `Docs` section from the Rust doc, fixed: https://github.com/iotaledger/iota/pull/5735/commits/99c8eee16590682c1b5e01bc02b36fa9499c595e#diff-fc30500a6b8c33380223c735cb9d6786afa2cc39301199f614993173f3acdb85L34-L36